src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -516,8 +517,49 @@ bool AppHost::HasWindow()
     return _shouldCreateWindow;
 }
 
+// Method Description:
+// - Event handler for the Peasant::ExecuteCommandlineRequested event. Take the
+//   provided commandline args, and attempt to parse them and perform the
+//   actions immediately. The parsing is performed by AppLogic.
+// - This is invoked when another wt.exe instance runs something like `wt -w 1
+//   new-tab`, and the Monarch delegates the commandline to this instance.
+// Arguments:
+// - args: the bundle of a commandline and working directory to use for this invocation.
+// Return Value:
+// - <none>
 void AppHost::_DispatchCommandline(winrt::Windows::Foundation::IInspectable /*sender*/,
-                                   winrt::Microsoft::Terminal::Remoting::CommandlineArgs args)
+                                   Remoting::CommandlineArgs args)
 {
-    _logic.ExecuteCommandline(args.Args());
+    _logic.ExecuteCommandline(args.Args(), args.CurrentDirectory());
+}
+
+// Method Description:
+// - Event handler for the WindowManager::FindTargetWindowRequested event. The
+//   manager will ask us how to figure out what the target window is for a set
+//   of commandline arguments. We'll take those arguments, and ask AppLogic to
+//   parse them for us. We'll then set ResultTargetWindow in the given args, so
+//   the sender can use that result.
+// Arguments:
+// - args: the bundle of a commandline and working directory to find the correct target window for.
+// Return Value:
+// - <none>
+void AppHost::_FindTargetWindow(const winrt::Windows::Foundation::IInspectable& /*sender*/,
+                                const Remoting::FindTargetWindowArgs& args)
+{
+    const auto targetWindow = _logic.FindTargetWindow(args.Args().Args());

```

Comment:
Care to find a better naming to avoid `args.Args().Args()` ? I mean, haha. I can't help myself but noticed this 😅